src/arraystring.rs

```diff
@@ -649,12 +664,11 @@ impl<const N: usize> ArrayString<N> {
     /// ```
     #[inline]
     pub fn push_str<S>(&mut self, string: S)
-    where
-        S: AsRef<str>,
+        where
+            S: AsRef<str>,
     {
         trace!("Push str truncate");
-        let size = Self::capacity().saturating_sub(self.len());
-        unsafe { self.push_str_unchecked(truncate_str(string.as_ref(), size)) }
+        let _ = self.try_push_str(truncate_str(string.as_ref(), self.remaining().len().into_lossy()));
```

Comment:
Why use try_push, that has an extra check when we already truncate it? Specially ignoring the error

Reply:
It's to phase out the unchecked version. It's actually as fast as with the unchecked version except for smallstring where it has a 10% overhead (200 ps). You can see for yourself, i updated the benches to prevent parameter optimization.

Reply:
Cool, I like this idea!